02week/pigLatin.js

```diff
@@ -3,49 +3,136 @@
 const assert = require('assert');
 const readline = require('readline');
 const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
+    input: process.stdin,
+    output: process.stdout
 });
 
 
 function pigLatin(word) {
 
-  // Your code here
+    // Your code here
+    const arrayVowels = ['a', 'e', 'i', 'o', 'u'];
+    const arrayConsonants = ['b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n', 'p', 'q', 'r', 's', 't', 'v', 'w', 'x', 'y', 'z'];
+    const allLetters = arrayVowels.concat(arrayConsonants);
+
+    const wordLower = word.toLowerCase();
+    //need a method to throw an error if they enter anything other than arrayVowels or arrayConsonants
+
+    // if((typeof wordLower) !='string'){
+    // console.log('Invalid entry, please choose a word. Numbers and special characters are not allowed.')
+    // }
+    //trying to check if the entry is a string, if not, I want to display a message about using words. - update, that won't work
+
+    const splitWordArray = (wordLower).split('');
+    //spliting the word the user entered into an splitWordArray
+//need a method to throw an error if they enter anything other than arrayVowels or arrayConsonants
+    for (let i = 0 ; i<splitWordArray.length; i++) {
+        if (!allLetters.includes(splitWordArray[i])) {
+          console.log('Invalid entry, please choose a word. Numbers and special characters are not allowed.');
+          return;
+        }
+    }
+          //should ask someone about best practices of leaving a blank if, and a filled else
+
+    //console.log(splitWordArray);
+    //this is to test that the word split function is operating
+
+
+    // const isVowel = ['a','e','i','o','u'];
+    // let isVowel = 'a'||'e'||'i'||'o'||'u';
+    //can I set a variable to be equal to this or that?
+    //went with the array and used the includes function to determine if vowel.
+
+    if (arrayVowels.includes(splitWordArray[0])) {
+        console.log(wordLower + 'way')
+    } else {
+        var vowelLocation = null;
+        for (let i = 1; i<splitWordArray.length; i++) {
```

Comment:
you are looping through the same array a lot and unnecessarily, what's a better way to simplify?